### DISCUSSION TITLE
Known bt status

### DISCUSSION THREAD

**Discussion by miguelavh** [zepp-health/zeppos-forum#349] [Q&A] [answered]
Hi is there a way to known if watch have bt connected programaticaly at watchface?

Another question, have any possibility to know api level at watchface?

Two questions have easy answer at mini-app but what about watchface?

Thank you for your help...

**Comment by SashaCX75** [comment, marked as answer]
Checking Bluetooth status in watch faces works the same way as in apps
https://docs.zepp.com/docs/1.0/reference/device-app-api/hmBle/

To check the API level, you need to use getSystemInfo
https://docs.zepp.com/docs/v2/reference/device-app-api/newAPI/settings/getSystemInfo/
This function works since API 2.1, so you need to add error handling when calling it. If the function call ends in an error, the device has API 1. Otherwise, the function will return the API value.

**Reply by miguelavh** [reply]
Thank you for your fast response.

BT check works perfect..., thank you

But API Level doesn't seems to work, @zos make code explode at Watchface, jajaja... do you know another approach for watchface?

Thank you in advance...

**Comment by SashaCX75** [comment]
You did not specify which API you use for watch faces. If you use API 1, the code should be modified as follows:
```
const systemInfo = hmSetting.getSystemInfo():
console.log(`osVersion = ${systemInfo.osVersion}`);
console.log(`firmwareVersion = ${systemInfo.firmwareVersion}`);
console.log(`minAPI = ${systemInfo.minAPI}`);
```

**Reply by miguelavh** [reply]
thank you but getSysemInfo() is not defined on hmSetting "Cannot read properties of undefined" (It's not documented at Watchface API page)... I'm testing Active with Zeppos 3.5, and making a Watchface, fetch-api is not working with Zeppos 3.0 to obtain data from URL, And I need to obtain osVersion to use fetch-api (Watchface too is valid for GTS3, GTS4 and  other watches with OS 2 and 2.1) or another type of update... Thanks anyway, If it's not possible I'll make a different watchface for each OS version...

**Comment by SashaCX75** [comment]
[SystemInfo_WF.zip](https://github.com/zepp-health/zeppos-forum/files/15444945/SystemInfo_WF.zip)
Here are 2 watch faces. One is made on API 1, the second on API 2. Both work fine. I checked on Balance and on Active simulator. So if something doesn't work for you, you must be doing something wrong. Or your watch is not working the same way as other models.

PS: fetch-api doesn't seem to work on watch faces. So I'm not sure how using fetch-api is related to watch faces.

**Reply by miguelavh** [reply]
Hi @SashaCX75 you are right, my mistake it works... I have ended the code line with ":" character... oh my god... In my defense, there is no reference in the API's doc to that function in hmSettings and it has made me suspects... Thank you very much.

**Reply by SashaCX75** [reply]
There was indeed a typo in my example and I wrote ":" at the end of the line. When writing code in VScode, this error was immediately highlighted.
Many functions specified only for applications can also work in watch faces. The main thing is that the watch should support such a function.

**Reply by miguelavh** [reply]
Sure I'll test VSCode, I'm coding with notepad++ (file by file... old school) and this kind of error code is not highlighted...
